io/pci/pci_hotplug.py

```diff
@@ -128,16 +129,71 @@ def is_recovered():
             `pre` value, and returns True if equals pre.
             False otherwise.
             """
-            post = len(pci.get_disks_in_pci_address(pci_addr))
-            post += len(pci.get_nics_in_pci_address(pci_addr))
-            self.log.debug("Pre: %d,  Post: %d",
-                           self.end_devices[pci_addr], post)
-            if post == self.end_devices[pci_addr]:
+            def is_path_online():
+                path_stat = []
+                path_stat = list(multipath.get_path_status(curr_path))
+                self.log.info("path=%s status=%s" % (curr_path, path_stat))
+                if path_stat[0] != 'active' or path_stat[2] != 'ready':
+                    return False
+                return True
+
+            curr_path = ''
```

Comment:
`curr_path = ''` is this needed ?

Reply:
yes, this is need as we are making this as global withing the main function.